src/theme-preval.js

```diff
@@ -48,58 +48,20 @@ const fontSizes = ['12px', '14px', '16px', '20px', '24px', '32px', '40px', '48px
 
 const space = ['0', '4px', '8px', '16px', '24px', '32px', '40px', '48px', '64px', '80px', '96px', '112px', '128px']
 
-const light = partitionColors(primitives.colors.light)
-const dark = partitionColors(primitives.colors.dark)
-const darkDimmed = partitionColors(primitives.colors['dark_dimmed'])
-const darkHighContrast = partitionColors(primitives.colors['dark_high_contrast'])
-
-// This file must be in vanilla JS to work with preval
-// but our temporary filter utils make it impossible for
-// our TypeScript to properly infer const object structure
-// so we need to use JSDoc comments.
-
-/**
- * @type Partial<typeof primitives.colors.light>
- */
-const lightColors = omitScale(light.colors)
-
-/**
- * @type Partial<typeof primitives.colors.light>
- */
-const lightShadows = omitScale(light.shadows)
-
-/**
- * @type Partial<typeof primitives.colors.dark>
- */
-const darkColors = omitScale(dark.colors)
-
-/**
- * @type Partial<typeof primitives.colors.dark>
- */
-const darkShadows = omitScale(dark.shadows)
-
-/**
- * @type Partial<typeof primitives.colors.dark_dimmed>
- */
-const darkDimmedColors = omitScale(darkDimmed.colors)
-
-/**
- * @type Partial<typeof primitives.colors.dark_dimmed>
- */
-const darkDimmedShadows = omitScale(darkDimmed.shadows)
-
-/**
- * @type Partial<typeof primitives.colors.dark_dimmed>
- */
-const darkHighContrastColors = omitScale(darkHighContrast.colors)
-
 /**
- * @type Partial<typeof primitives.colors.dark_high_contrast>
+ * @type Record<keyof typeof primitives.colors, Record<'colors' | 'shadows', Partial<typeof primitives.colors.light>>
  */
-const darkHighContrastShadows = omitScale(darkHighContrast.shadows)
+const colorSchemes = Object.entries(primitives.colors)
+  .map(([name, variables]) => {
+    const {colors, shadows} = partitionColors(variables)
+    return [name, {colors: omitScale(colors), shadows: omitScale(shadows)}]
+  })
+  .reduce((acc, [name, variables]) => {
+    acc[name] = variables
+    return acc
+  }, {})
```

Comment:
I refactored this to get all color schemes from Primer Primitives instead of hard-coding references to them. This means that when future color schemes are published in Primer Primitives, we only need to bump the Primer Primitives version number in Primer React to make the color schemes available to Primer React consumers.